pkg/controller/networkpolicy/antreanetworkpolicy.go

```diff
@@ -123,38 +124,57 @@ func (n *NetworkPolicyController) deleteANP(old interface{}) {
 // in case of ADD event or modified and store the updated instance, in case
 // of an UPDATE event.
 func (n *NetworkPolicyController) processAntreaNetworkPolicy(np *secv1alpha1.NetworkPolicy) *antreatypes.NetworkPolicy {
-	appliedToGroupNames := make([]string, 0, len(np.Spec.AppliedTo))
-	// Create AppliedToGroup for each AppliedTo present in
-	// AntreaNetworkPolicy spec.
+	appliedToPerRule := np.Spec.AppliedTo == nil
```

Comment:
is this going to be nil? or []?

Reply:
This is going to be nil, but per Quan's comments, changed to len(AppliedTo) to be safe. Thanks for suggesting.

Reply:
probably my golang knowledge is not that deep, is ranging over a nil value allowed?